wgpu-core/src/pipeline.rs

```diff
@@ -113,6 +113,10 @@ pub enum CreateComputePipelineError {
     Implicit(#[from] ImplicitLayoutError),
     #[error(transparent)]
     Stage(validation::StageError),
+    #[error("Internal error: {0}")]
+    Internal(String),
+    #[error("not enough memory left")]
+    OutOfMemory,
```

Comment:
let's keep the old way of returning OOM via `DeviceError`, no need to add this variant here

Reply:
Could you explain, how I can defeat the compiler in this way?

```rust
error[E0308]: `match` arms have incompatible types
    --> wgpu-core/src/device/mod.rs:2399:25
     |
2396 |                   match err {
     |                   --------- `match` arms have incompatible types
2397 |                       hal::pso::CreationError::OutOfMemory(_) => DeviceError::OutOfMemory,
     |                                                                  ------------------------ this is found to be of type `DeviceError`
2398 |                       hal::pso::CreationError::ShaderCreationError(stage, error) => {
2399 | /                         pipeline::CreateRenderPipelineError::Internal {
2400 | |                             stage: conv::map_hal_flags_to_shader_stage(stage),
2401 | |                             error,
2402 | |                         }
     | |_________________________^ expected enum `DeviceError`, found enum `CreateRenderPipelineError`
```

Reply:
You'd do:
```rust
hal::pso::CreationError::OutOfMemory(_) => hal::pso::CreationError::ShaderCreationError::Device(DeviceError::OutOfMemory),
```

Reply:
TIL. Thank you!